src/classes/HttpsProxyAgent.js

```diff
@@ -29,8 +29,7 @@ class HttpsProxyAgent extends Agent {
 
     socket.once('data', () => {
       const secureSocket = tls.connect({
-        rejectUnauthorized: false,
```

Comment:
Note: I wonder why `rejectUnauthorized` was hardcoded here with `false` as default value.
This is quite insecure and should be `true` by default, and could be overridden with custom request's option.

Reply:
Well, unfortunately there are no test cases for it & I cannot remember. Will merge it as is and see if anything breaks.